cli/command/logs/follow.go

```diff
@@ -1,9 +1,107 @@
 package logs
 
 import (
+	"fmt"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/gorilla/websocket"
 	"github.com/storageos/go-cli/cli/command"
+	"github.com/storageos/go-cli/cli/command/formatter"
+)
+
+const (
+
+	// Endpoint path for logs
+	logsPath = "/v1/logs"
+
+	// Time allowed to read the next message from the peer.
+	wait = 60 * time.Second
+
+	// Send pings to peer with this period. Must be less than wait.
+	pingPeriod = (wait * 9) / 10
+
+	// Maximum message size allowed from peer.
+	maxMessageSize = 4096
 )
 
 func runFollow(storageosCli *command.StorageOSCli, opt logOptions) error {
-	return nil
+
+	format := opt.format
+	if len(format) == 0 {
+		format = formatter.TableFormatKey
+	}
+
+	fmtCtx := formatter.Context{
+		Output: storageosCli.Out(),
+		Format: formatter.NewLogStreamFormat(format, opt.quiet),
+	}
+
+	c, err := storageosCli.WebsocketConn(logsPath)
+	if err != nil || c == nil {
+		return fmt.Errorf("Connection error: %v", err)
+	}
+	defer c.Close()
+
+	c.SetPongHandler(func(string) error { return c.SetReadDeadline(time.Now().Add(wait)) })
+	c.SetReadLimit(maxMessageSize)
+
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
+
+	done := make(chan struct{})
+
+	// Incoming message handler
```

Comment:
This is fine but keeping the run functions small helps in isolating the main code and adding test to validate the correctness of the code, wherever possible. How about moving this handlers out as separate function?